android/src/firebase/core/TitaniumFirebaseCoreModule.java

```diff
@@ -31,112 +31,132 @@
 	private static final boolean DBG = TiConfig.LOGD;
 
-	public TitaniumFirebaseCoreModule()
-	{
+	public TitaniumFirebaseCoreModule() {
 		super();
 	}
 
 	// Public APIs
 
-	@Kroll.method
-	public void configure(@Kroll.argument(optional = true) KrollDict param)
-	{
-		if (param != null) {
-			String apiKey = "";
-			String databaseURL = "";
-			String projectID = "";
-			String storageBucket = "";
-			String applicationID = "";
-			String GCMSenderID = "";
-			FirebaseOptions.Builder options = new FirebaseOptions.Builder();
-
-			if (param.containsKey("file")) {
-				// open file and parse it
-				try {
-					JSONObject json = new JSONObject(loadJSONFromAsset(param.getString("file")));
-					JSONObject projectInfo = json.getJSONObject("project_info");
-					String packageName = TiApplication.getAppCurrentActivity().getPackageName();
+	public final String DEFAULTJSONFILE = "google-services.json";
 
-					if (projectInfo.has("storage_bucket")) {
-						storageBucket = projectInfo.getString("storage_bucket");
-					}
-					if (projectInfo.has("firebase_url")) {
-						databaseURL = projectInfo.getString("firebase_url");
-					}
-					if (projectInfo.has("project_number")) {
-						GCMSenderID = projectInfo.getString("project_number");
-					}
-					if (projectInfo.has("project_id")) {
-						projectID = projectInfo.getString("project_id");
-					}
-					if (json.has("client")) {
-						JSONArray clients = json.getJSONArray("client");
-						for (int i = 0, len = clients.length(); i < len; i++) {
-							JSONObject client = clients.getJSONObject(i);
-							JSONObject clientInfo = client.getJSONObject("client_info");
-							String pName = clientInfo.getJSONObject("android_client_info").getString("package_name");
-							if (pName.equals(packageName)) {
-								applicationID = client.getJSONObject("client_info").getString("mobilesdk_app_id");
-								apiKey = client.getJSONArray("api_key").getJSONObject(0).getString("current_key");
-							}
-						}
-					}
-				} catch (JSONException e) {
-					Log.e(LCAT, "Error parsing file");
-				}
-			} else {
-				// use parameters
-				if (param.containsKey("APIKey")) {
-					apiKey = param.getString("APIKey");
-				}
-				if (param.containsKey("databaseURL")) {
-					databaseURL = param.getString("databaseURL");
+	@Kroll.method
+	public void configure(@Kroll.argument(optional = true) Object object) {
+		String filenameOfJSON = null;
+		String apiKey = "";
+		String databaseURL = "";
+		String projectID = "";
+		String storageBucket = "";
+		String applicationID = "";
+		String GCMSenderID = "";
+		Log.d(LCAT,"Start reading of firebase paramters:\n======================");
+		if (object == null) {
+			// if args empty we use default filename.
+			// we cannot use the standard native way, because the default place of file
+			// is unrechable by titanium 
+			Log.i(LCAT, "configure parameter was empty, use " + DEFAULTJSONFILE);
+			filenameOfJSON = DEFAULTJSONFILE;
+		} else if (object instanceof String) {
```

Comment:
please keep the `param.containsKey("file")` since iOS is using `file` too. And then we could keep the `KrollDict param` as a parameter for the configure function.